meta/data.go

```diff
@@ -177,6 +177,43 @@ func (data *Data) DropDatabase(name string) error {
 	return ErrDatabaseNotFound
 }
 
+// RenameDatabase renames a database.
+// Returns an error if oldName or newName is blank
+// or if a database with the newName already exists
+// or if a database with oldName does not exist
+func (data *Data) RenameDatabase(oldName, newName string) error {
+	if newName == "" || oldName == "" {
+		return ErrDatabaseNameRequired
+	}
+	if data.Database(newName) != nil {
+		return ErrDatabaseExists
+	}
+	if data.Database(oldName) == nil {
+		return ErrDatabaseNotFound
+	}
+	// find database named oldName and rename it to newName
+	for i := range data.Databases {
+		if data.Databases[i].Name == oldName {
+			data.Databases[i].Name = newName
+			data.switchDatabaseUserPrivileges(oldName, newName)
+			// TODO should rename the databases used in continuous queries
```

Comment:
@otoolep  --- OK-- I see why it would be good to error at this point then.

Would it be sufficient to check all the CQs and throw the error if the 
query contains " ON <olddb> " or " INTO \"<olddb>\"."  or " FROM \"<olddb>\"." ?

I guess the error should be a new one (ErrDatabaseRenameCQConflict ?)
What do you think?